staging/src/k8s.io/cloud-provider/options/webhook.go

```diff
@@ -68,6 +85,30 @@ func (o *WebhookOptions) Validate(allWebhooks, disabledByDefaultWebhooks []strin
 			allErrors = append(allErrors, fmt.Errorf("%q is not in the list of known webhooks", webhook))
 		}
 	}
+	if len(o.Webhooks) != 0 && o.ValidatingWebhookConfigFilePath == "" {
+		allErrors = append(allErrors, errors.New("webhooks are enabled but the webhook configuration path is empty"))
+	}
+	if o.validationWebhookConfiguration != nil {
+		if o.validationWebhookConfiguration.Name == "" {
+			allErrors = append(allErrors, errors.New("validating webhook configuration name can't be empty"))
+		}
+		webhookConfigs := sets.NewString()
+		for _, webhookConfig := range o.validationWebhookConfiguration.Webhooks {
+			webhookConfigs.Insert(webhookConfig.Name)
+		}
+		for _, name := range allWebhooks {
+			if genericcontrollermanager.IsControllerEnabled(name, sets.NewString(disabledByDefaultWebhooks...), o.Webhooks) {
+				if !webhookConfigs.Has(name) {
+					allErrors = append(allErrors, fmt.Errorf("webhook %s is enabled but is not present in the webhook configuration", name))
+				} else {
+					webhookConfigs.Delete(name)
+				}
+			}
+		}
+		if webhookConfigs.Len() != 0 {
```

Comment:
So the behavior here is that we need a 1:1 mapping of enabled webhooks to webhook configs, right?  I think that makes sense.  The only case I'm thinking of is where a user wants to disable the webhook and reenable it later, but I think its better to force them to remove the config because that way its explicit what webhooks exist based on the configuration and the webhook names.

Reply:
yes, that correct. Since both are configs provided by the user i felt its better to force to update both places.